### ISSUE TITLE
babelify 7.2.0 change the way source map path is generated

### ISSUE THREAD

**Issue by osmanpontes** [babel/babelify#174]
When I use babelify 6.4.0, I have this inside my source map file:
```
"sources": [
    "node_modules/browser-pack/_prelude.js",
    "C:/Users/Osman/Dropbox/react-flux-init/client.js",
    "C:/Users/Osman/Dropbox/react-flux-init/components/About.jsx",
    "C:/Users/Osman/Dropbox/react-flux-init/components/App.jsx",
    "C:/Users/Osman/Dropbox/react-flux-init/components/Index.jsx",
    "node_modules/deep-equal/index.js",
    "node_modules/deep-equal/lib/is_arguments.js",
    "node_modules/deep-equal/lib/keys.js",
    "node_modules/history/lib/Actions.js",
    "node_modules/history/lib/AsyncUtils.js",
    "node_modules/history/lib/DOMStateStorage.js",
    "node_modules/history/lib/DOMUtils.js",
    "node_modules/history/lib/ExecutionEnvironment.js",
    "node_modules/history/lib/createBrowserHistory.js",
    "node_modules/history/lib/createDOMHistory.js",
    "node_modules/history/lib/createHistory.js",
    "node_modules/history/lib/createLocation.js",
    "node_modules/history/lib/deprecate.js",
    "node_modules/history/lib/extractPath.js",
    "node_modules/history/lib/parsePath.js",
    "node_modules/history/lib/runTransitionHook.js",
    "node_modules/invariant/browser.js",
    "node_modules/process/browser.js",
    "node_modules/warning/browser.js",
    "C:/Users/Osman/Dropbox/react-flux-init/routes.js"
  ],
```

And when I use babelify 7.2.0, I have this:
```
"sources": [
    "node_modules/browser-pack/_prelude.js",
    "client.js",
    "components\\About.jsx",
    "components\\App.jsx",
    "components\\Index.jsx",
    "node_modules/deep-equal/index.js",
    "node_modules/deep-equal/lib/is_arguments.js",
    "node_modules/deep-equal/lib/keys.js",
    "node_modules/history/lib/Actions.js",
    "node_modules/history/lib/AsyncUtils.js",
    "node_modules/history/lib/DOMStateStorage.js",
    "node_modules/history/lib/DOMUtils.js",
    "node_modules/history/lib/ExecutionEnvironment.js",
    "node_modules/history/lib/createBrowserHistory.js",
    "node_modules/history/lib/createDOMHistory.js",
    "node_modules/history/lib/createHistory.js",
    "node_modules/history/lib/createLocation.js",
    "node_modules/history/lib/deprecate.js",
    "node_modules/history/lib/extractPath.js",
    "node_modules/history/lib/parsePath.js",
    "node_modules/history/lib/runTransitionHook.js",
    "node_modules/invariant/browser.js",
    "node_modules/process/browser.js",
    "node_modules/warning/browser.js",
    "routes.js"
  ],
```

Using babelify 6.4.0 the result in chrome dev tools is:

![image](https://cloud.githubusercontent.com/assets/1644019/12401138/abde9dc2-be0b-11e5-9349-66ba68142983.png)

Using babelify 7.2.0 the result in chrome dev tools is:

![image](https://cloud.githubusercontent.com/assets/1644019/12401229/4fedc24e-be0c-11e5-8026-ac86804fdca4.png)

The problem is that I have **components\About.jsx** like the name of the file instead of the file **About.jsx** inside a folder structure.

In the source map files, we can see that in babelify 6.4.0 we have `C:/Users/Osman/Dropbox/react-flux-init/components/About.jsx` and in babelify 7.2.0 we have `components\\About.jsx`.
I bet the problem is the use of `\\` instead of `/` to make folder path. Other thing to notice is that the first is full path, and the second is relative to project folder.

Why babelify change this behavior? Is it a bug? What should I do to have the first behavior in babelify 7.2.0? It wasn't easy to find this. I searched a lot.

Ps1.: I used exorcist package to take the source map in external file.

Ps2.: I think this is a very specific question, so I prefered post here instead of stackoverflow.

**Comment by zertosh** [comment]
Are you using the `sourceMapRelative` option? If so, try turning it off.

**Comment by osmanpontes** [comment]
No, I'm not using this option.

The package.json using babelify 6.4.0 is:

```
{
  "name": "react-flux-init",
  "version": "0.0.0",
  "main": "index.js",
  "scripts": {
    "vendor": "browserify -r react -r react-dom -r react-router > public/vendor.js",
    "app": "browserify -x react -x react-dom -x react-router client.js -d | exorcist public/app.js.map > public/app.js"
  },
  "keywords": [],
  "author": "Osman Pontes",
  "license": "ISC",
  "description": "",
  "dependencies": {
    "history": "^1.17.0",
    "react": "^0.14.6",
    "react-dom": "^0.14.6",
    "react-router": "^1.0.3"
  },
  "devDependencies": {
    "babel-cli": "^6.4.0",
    "babel-preset-es2015": "^6.3.13",
    "babel-preset-react": "^6.3.13",
    "babelify": "^6.4.0",
    "browserify": "^13.0.0",
    "exorcist": "^0.4.0",
    "watchify": "^3.7.0"
  },
  "browserify": {
    "transform": [
      [
        "babelify",
        {
          "blacklist": [
            "regenerator"
          ]
        }
      ]
    ]
  }
}
```

And using babelify 7.2.0 is:

```
{
  "name": "react-flux-init",
  "version": "0.0.0",
  "main": "index.js",
  "scripts": {
    "vendor": "browserify -r react -r react-dom -r react-router > public/vendor.js",
    "app": "browserify -x react -x react-dom -x react-router client.js -d | exorcist public/app.js.map > public/app.js"
  },
  "keywords": [],
  "author": "Osman Pontes",
  "license": "ISC",
  "description": "",
  "dependencies": {
    "history": "^1.17.0",
    "react": "^0.14.6",
    "react-dom": "^0.14.6",
    "react-router": "^1.0.3"
  },
  "devDependencies": {
    "babel-cli": "^6.4.0",
    "babel-preset-es2015": "^6.3.13",
    "babel-preset-react": "^6.3.13",
    "babelify": "^7.2.0",
    "browserify": "^13.0.0",
    "exorcist": "^0.4.0",
    "watchify": "^3.7.0"
  },
  "browserify": {
    "transform": [
      [
        "babelify",
        {
          "presets": [
            "es2015",
            "react"
          ]
        }
      ]
    ]
  }
}

```

**Comment by kuraga** [comment]
+1, the same issue

**Comment by kuraga** [comment]
P.S. https://github.com/babel/babelify/commit/6ca571cf65457ed1c935f04c4cfd7adcd0b91984#diff-1a5f342c044db9fae335a88325457004R17

**Comment by davelandry** [comment]
:+1:

I'm encountering this issue when using [coverify](https://github.com/substack/coverify)

**Comment by zertosh** [comment]
I've published babelify@7.3.0. There's no `sourceMapRelative` option anymore. Instead, there's a `sourceMapsAbsolute` that you can set to get the old behavior.

**Comment by kuraga** [comment]
Thanks very much! Confirm: it works.
